app/pages/patientdata/patientdata.js

```diff
@@ -590,18 +592,20 @@ export let PatientData = React.createClass({
 
   componentWillReceiveProps: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
-    var currentPatientData = _.get(this.props, ['patientDataMap', userId], null);
-
     var nextPatientData = _.get(nextProps, ['patientDataMap', userId], null);
 
-    if (!currentPatientData && nextPatientData) {
+    // Hold processing until patient is fetched (ensuring settings are accessible), AND
+    // processing hasn't already taken place (this should be cleared already when switching patients), AND
+    // nextProps patient data exists
+    if (!nextProps.fetchingPatient && !this.state.processedPatientData && nextPatientData) {
       this.doProcessing(nextProps);
     }
   },
 
   doProcessing: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
     var patientData = _.get(nextProps, ['patientDataMap', userId], null);
+    var patientSettings = _.get(nextProps, ['patient', 'settings'], DEFAULT_SETTINGS);
```

Comment:
For posterity, I also considered updating [this area](https://github.com/tidepool-org/blip/blob/3a89fe4a8e84e0beef24e8b33497bcf51951e4aa/app/core/api.js#L347), to set `person.settings = settings || *null*` instead of `{}`, and kept the above as-is. But that could affect the site change source code, and I didn't want to open that can of worms again. Might be something to consider in the future, once `platform` takes over the metadata endpoints, though!